### DISCUSSION TITLE
Is Noscript necessary if I am running Firefox inside firejail ?

### DISCUSSION THREAD

**Discussion by firejailaddssecuirty** [netblue30/firejail#4801] [Q&A] [answered]
Frankly browsing the web with Noscript is a tiresome job. Is Noscript really necessary if I am running Firefox inside firejail ?

**Comment by topimiettinen** [comment, marked as answer]
I'd say yes. Noscript and Firejail operate at different level. Not running unknown Javascript may prevent hardware level vulnerabilites from being exploited, such as [Rowhammer](https://en.wikipedia.org/wiki/Row_hammer), [TLBleed](https://en.wikipedia.org/wiki/TLBleed) and [Spectre](https://en.wikipedia.org/wiki/Spectre_(security_vulnerability)) variants, as well as vulnerabilities in the Firefox which are accessible from JS. Firejail, AppArmor/SELinux or other kernel protections can't prevent those, since the vulnerabilities aren't in the interfaces which they protect.

**Reply by firejailaddssecuirty** [reply]
In that case I will keep using Noscript no matter how problematic it is. Thanks.

**Reply by firejailaddssecuirty** [reply]
> Not running unknown Javascript may prevent hardware level vulnerabilites from being exploited

I have a question. You said "hardware level vulnerabilities". A firejail sandbox for Firefox does not offer protection against hardware level vulnerabilities ? So all firejail does is isolate Firefox so that user's personal data leaving ~/Downloads is not accessible ?

**Reply by topimiettinen** [reply]
> > Not running unknown Javascript may prevent hardware level vulnerabilites from being exploited
> 
> I have a question. You said "hardware level vulnerabilities". A firejail sandbox for Firefox does not offer protection against hardware level vulnerabilities ?

No. If there was a way, we'd surely try to use it. Kernel may for example flush caches or turn on hardware mitigations from CPU manufacturers ([SSBD](https://en.wikipedia.org/wiki/CPUID#SSBD), IBRS, IBPB etc.) but I don't think they offer very good protection.

> So all firejail does is isolate Firefox so that user's personal data leaving ~/Downloads is not accessible ?

Firejail uses several methods (maybe even every method made available by kernel, except Landlock and KVM) to make attackers' lives more difficult: seccomp, mount/network/pid namespaces and capabilities. Still, hostile JS could use Rowhammer, Spectre & co. to attack the kernel directly. If that approach is successful and the attacker can read from or write to kernel memory or even execute code at kernel privilege level, nothing Firejail does would matter much.

**Comment by rusty-snake** [comment]
> Frankly browsing the web with Noscript is a tiresome job. 

OT: Use ublock origin in hard/medium mode and fallback to medium/easy mode if necessary.

**Reply by firejailaddssecuirty** [reply]
![addons](https://user-images.githubusercontent.com/93089434/147713297-24dba905-92bf-41d6-b46f-f35477a7dbe3.png)
I am already using uBlock Origin. Without uBlock origin I see all kinds of annoying ads. Noscript does nothing to stop them from appearing. These are the addons that I am using (please see attachment). I went into the uBlock Origin settings but couldn't find  hard/medium mode.

**Reply by rusty-snake** [reply]
Official documentation: https://github.com/gorhill/ublock/wiki/Blocking-mode

[My suggestion](https://codeberg.org/rusty-snake/firefox-config):
- Block JS
- Block 3p and 3p-script
- Make frames click-to-load

Then you can allow JS if required, allow 3p(-script) if required, ...

---

> HTTPS Everywhere

is deprecated, just use HTTPS-only-mode.

**Reply by firejailaddssecuirty** [reply]
I tried the blocking mode of uBlockOrigin & it felt exactly the same as Noscript. Pages broke unless I allowed stuff which again is what I am already doing under Noscript. So the conclusion is both privacy & security comes at a cost. I am not sure about what I am going to say but it seems as if people who have cash for a VPN can get away without Noscript. 

Removed HTTPS Everywhere & enabled HTTPS-only-mode.

**Reply by rusty-snake** [reply]
> I am not sure about what I am going to say but it seems as if people who have cash for a VPN can get away without Noscript.

apples and pears

VPNs and NoScript have nothing to do with each other.

**Reply by firejailaddssecuirty** [reply]
[https://www.youtube.com/watch?v=iPRpxze4eH0](url)
This video says javascript are used by advertisers for tracking users. So if someone is using a VPN the advertisers doesn't get to know the users's real IP so can't track him/her. You disagree ?

**Reply by rusty-snake** [reply]
1. This is one reason why you should disable JS, there may be more (security for example.
2. IP addresses are much more dynamic today (mobile internet) than ten years ago and are used less for tracking users. If you use a VPN but don't clean cookies you can be tracked.
3. JS is required for (most) active FPing.

**Reply by firejailaddssecuirty** [reply]
About cleaning cookies, I am using the arkenfox custom user.js & it cleans cookies as soon as I close Firefox. I searched the term "FPing" and found 
>fping is a program like ping which uses the Internet Control Message Protocol ( ICMP ) echo request to determine if a target host is responding

Did you meant this one or some other FPing ?  How are javascripts and FPing related ?

**Reply by rusty-snake** [reply]
FP = FingerPrint
